src/test/java/testtype/CVSB_3942/TestCertificateTir_CVSB_3942.java

```diff
@@ -0,0 +1,194 @@
+        testTypeDetailsSteps.sendCertificateNumber("1234");
+        testTypeDetailsSteps.pressSave();
+        testSteps.checkTestTypeStatus("Technical test", TestPage.TestTypeStatuses.IN_PROGRESS);
+        testSteps.clickReviewAndSubmit();
+        testSteps.checkErrorMessageMandatoryFieldsDisplayed();
+        testSteps.selectTestType("Technical test", TestPage.TestTypeStatuses.IN_PROGRESS);
+        testTypeDetailsSteps.checkCertificateNumberIs("1234");
+        testTypeDetailsSteps.checkCertificateErrorIsDisplayed();
+        testTypeDetailsSteps.sendCertificateNumber("12345");
+        testTypeDetailsSteps.checkCertificateErrorIsNotDisplayed();
+        testTypeDetailsSteps.pressSave();
+        testSteps.checkTestTypeStatus("Technical test", TestPage.TestTypeStatuses.EDIT);
+        testSteps.clickReviewAndSubmit();
+        testReviewSteps.checkTestStatus("TIR test", "PASS");
+        testReviewSteps.checkElementIsDisplayed("Certificate number");
+        testReviewSteps.checkElementValue("Certificate number", "GB/V12345");
+        testReviewSteps.goToTestPage();
+        testSteps.selectTestType("Technical test", TestPage.TestTypeStatuses.EDIT);
+        testTypeDetailsSteps.scrollTestTypeDetailsPage();
+        testTypeDetailsSteps.pressTestTypeAbandonButton();
+        selectReasonSteps.checkReasonsTirList();
+        selectReasonSteps.selectTheReason(SelectReasonPage.ReasonsTir.REASON_4);
+        selectReasonSteps.pressNextButton();
+        abandonTestSteps.addAdditionalComments("testing purposes");
+        abandonTestSteps.pressDone();
+        abandonTestSteps.pressAbandon();
+        testSteps.checkTestTypeStatus("Technical test", TestPage.TestTypeStatuses.ABANDONED);
+        testSteps.addLinkedTestType();
+        testTypeCategorySteps.selectFromTestTypeList("Retest");
+        testTypeSubcategorySteps.selectFromTestTypeList("TIR");
+        testTypeSubcategorySteps.selectFromTestTypeList("Paid");
+        testSteps.checkTestTypeStatus("Retest", TestPage.TestTypeStatuses.IN_PROGRESS);
+        testSteps.selectTestType("Retest", TestPage.TestTypeStatuses.IN_PROGRESS);
+        testTypeDetailsSteps.setTestToOption("Pass");
+        testTypeDetailsSteps.checkCertificateNumberIsDisplayed();
+        testTypeDetailsSteps.sendCertificateNumber("54321");
+        testTypeDetailsSteps.pressSave();
+        testSteps.checkTestTypeStatus("Retest", TestPage.TestTypeStatuses.EDIT);
+        testSteps.clickReviewAndSubmit();
+        testReviewSteps.checkTestStatus("TIR test", "ABANDONED");
+        testReviewSteps.checkElementValue("Reason(s) for abandoning", selectReasonSteps.getReasonTexts(SelectReasonPage.ReasonsTir.REASON_4));
+        testReviewSteps.checkElementValue("Notes", "testing purposes");
+        testReviewSteps.scrollDown();
+        testReviewSteps.checkTestStatus("Paid TIR retest", "PASS");
+        testReviewSteps.checkElementValue("Certificate number", "CB/V54321");
+//        testReviewSteps.pressSubmit();
```

Comment:
Why are these lines commented out?

Reply:
submitting was beyond the scope of the test and if I recall correctly Sam suggested it some time ago, to reduce execution time